workflow/cron/operator.go

```diff
@@ -116,30 +121,26 @@ func getWorkflowObjectReference(wf *v1alpha1.Workflow, runWf *v1alpha1.Workflow)
 }
 
 func (woc *cronWfOperationCtx) persistUpdate() {
-	if woc.origCronWf.ResourceVersion != woc.cronWf.ResourceVersion {
-		woc.log.Error("cannot update cron workflow with mismatched resource versions")
+	if !woc.updated {
 		return
 	}
-	cronWf, err := woc.cronWfIf.Update(woc.cronWf)
+
+	_, err := woc.cronWfIf.Update(woc.cronWf)
 	if err != nil {
 		if !errors.IsConflict(err) {
 			woc.log.WithError(err).Error("failed to update CronWorkflow")
 			return
 		}
 		var reapplyErr error
-		cronWf, reapplyErr = woc.reapplyUpdate()
+		_, reapplyErr = woc.reapplyUpdate()
 		if err != nil {
 			woc.log.WithError(reapplyErr).WithField("original error", err).Error("failed to update CronWorkflow after reapply attempt")
 			return
 		}
 	}
-	woc.cronWf = cronWf
 }
 
 func (woc *cronWfOperationCtx) reapplyUpdate() (*v1alpha1.CronWorkflow, error) {
-	if woc.origCronWf.ResourceVersion != woc.cronWf.ResourceVersion {
```

Comment:
if you delete this check - you will have failing cron workflows (the will not be able to `reapplyUpdates` - so instead make sure the code problem is fixed

Reply:
After my change this check will be moot since `woc.cronWf.ResourceVersion` is never updated and will always be the same after `woc.origCronWf.ResourceVersion`.

The core idea in this change is that a single operation of the `cronWfOperationctx` is only able to persist the CRD one time, just as is done with the workflow operator.

> if you delete this check - you will have failing cron workflows (the will not be able to reapplyUpdates

Sorry, it's not obvious to me why removing the check will prevent cron workflows from running `reapplyUpdates`. Could you elaborate a bit more?

Reply:
This check is a "dynamic assertion". I.e. it checks for correctness at runtime. If it fails, we know we have a bug.

Theres is a check in `reapplyUpdates` - but this (should be) rarely called.

By keeping this check in place we prevent future bugs.

I like the use of `woc.updated` BTW.

Reply:
I see. Even though the check is moot now, it's in place to prevent bugs from future changes. I will add the checks back.